### PR TITLE
Improve MariaDB's connection

### PR DESCRIPTION
**Pull request by louislam**
Knex sometimes throws `Error: read ECONNRESET`. Trying to improve this.

Reference:
https://github.com/knex/knex/issues/3447#issuecomment-1320931175

### CODE REVIEW

**Comment by louislam** [comment]
Actually, I don't know how to solve this issue. Setting the pool doesn't seem to be working.

The issue only happens when the database connection is idle for too long.

**Comment by CommanderStorm** [comment]
> The issue only happens when the database connection is idle for too long

We could just create a no-op which is executed once every minute in a `setInterval` to keep the DB active, or am I missing something?

This issue should only happen for users who have not set up any monitors, right?

**Comment by louislam** [comment]
> > The issue only happens when the database connection is idle for too long
> 
> We could just create a no-op which is executed once every minute in a `setInterval` to keep the DB active, or am I missing something?
> 
> This issue should only happen for users who have not set up any monitors, right?

It can also affect MariaDB with custom timeout settings, also a very long interval monitor.

I just re-read the https://github.com/knex/knex/issues/3447#issuecomment-1320890276. It looks like I can override the `validate` function to prevent this.

File: server/database.js [hunk 1]
```diff
@@ -209,14 +209,9 @@ class Database {
         let config = {};
 
         let mariadbPoolConfig = {
-            pool: {
```
Comment (louislam) [inline server/database.js]: I added it wrongly, that might be the reason why `min: 0` was not working. Will test it again.

**Pull request merged**